jecs.d.ts

```diff
@@ -150,6 +150,14 @@ export class World {
 	 * @param component The component (or tag) to add.
 	 */
 	add<C>(entity: Entity, component: undefined extends InferComponent<C> ? C : Id<undefined>): void;
+	
+	/**
+	 * Assigns a value to a component on the given entity.
+	 * @param entity The target entity.
+	 * @param component The component definition (could be a Pair or Entity).
+	 * @param value The value to store with that component.
+	 */
+	set<E extends Id<unknown>>(entity: Entity, component: E, value: InferComponent<E>): void;
```

Comment:
You should probably be able to do `set<T>(entity: Entity, component: Id<T>, value: T)` which would simplify things. Make sure to test your results